src/persist-client/src/internal/compact.rs

```diff
@@ -268,6 +264,7 @@ where
     pub fn compact_and_apply_background(
         &self,
         req: CompactReq<T>,
+        machine: &Machine<K, V, T, D>,
```

Comment:
I think something in this file (or maybe the callsite) deserves a comment stating that we're intentionally passing a clone of the machine per req to maximize probability that the compaction res applies cleanly